app/models/accounting/quickbooks/transaction_extractor.rb

```diff
@@ -8,7 +8,33 @@ def initialize(txn)
         @loan = txn.project
       end
 
+      # TODO: if below order does not work for bill, purchase, or deposit,
+      # consider returning to this order:
+      # extract_account
+      # extract_additional_metadata
+      # extract_line_items
+      # set_managed
+      # calculate_amount
+      # add_implicit_line_items
+      # set_type
+      #
```

Comment:
The spec says:
```* calculate_amount: Set amount using current formula, *after* creating line items that mirror QB, but *before* child classes add additional line items! (This is because we need to build the line items that affect the receivable accounts in order to calculate the amount, and we need the amount in order to add the special line items.)
```
I'm not sure which step "add additional line items" is referring too. if If it's add_implicit_line_items then we're ok, but if it's something else we'll have revisit this.

Reply:
additional line items and implicit line items are the same thing, so we should be good